### PR TITLE
Oidc

### PR DESCRIPTION
**Pull request by louis-murray**

### CODE REVIEW

**Comment by louis-murray** [comment]
I have signed the cla

**Comment by louis-murray** [comment]
related to #294

**Comment by louis-murray** [comment]
ok, I had to set the openid-client package to version 2.5.0, same version used by the go daddy kube client, to support node 8.

File: src/oidc_auth.ts [hunk 1]
```diff
@@ -1,30 +1,66 @@
 import https = require('https');
+import { Client, Issuer } from 'openid-client';
 import request = require('request');
 
 import { Authenticator } from './auth';
 import { User } from './config_types';
 
 export class OpenIDConnectAuth implements Authenticator {
+    // public for testing purposes.
+    public currentTokenExpiration = 0;
```
[the remark this answers is not in the document]
Comment (louis-murray) [reply src/oidc_auth.ts]: that's slick, I'll make that change

**Comment by louis-murray** [comment]
just updated based on your comments, I also added the updated types for oidc and replaced the call when setting the `currentTokenExpiration` from `newToken.expiration` to `newToken.expires_at` because I realized the oidc spec defines `expires_at`as the standard and `expiration` is just something that is implementation dependent

**Comment by brendandburns** [comment]
/lgtm
/approve

Thanks for the patience!

**Pull request merged**